File: le1/le1/plp/expressions1/Exemplo1.java
package le1.plp.expressions1;
import le1.plp.expressions1.expression.*;


public class Exemplo1{
    public static void main(String[] args) {
        
        Expressao expr = new ExpSoma(new ValorString("abc"), new ExpSub((new ValorInteiro(12)), new ValorInteiro(5)));
        Programa programa = new Programa(expr);
        
        if (programa.checaTipo()) {
            System.out.println("Tipo válido!");
            programa.executar();
        } else {
            System.out.println("Erro de tipo!");
        }
    }
}

File: python/le1/plp/expressions1/Exemplo1.py
"""
Atividade 1 (Lab 01 - le1).

Avalia a expressao: 4 + 12 - 3
"""
from le1.plp.expressions1.Programa import Programa
from le1.plp.expressions1.expression.ExpSoma import ExpSoma
from le1.plp.expressions1.expression.ExpSub import ExpSub
from le1.plp.expressions1.expression.ValorInteiro import ValorInteiro


class Exemplo1:

    @staticmethod
    def main():
        # 4 + 12 - 3
        expressao = ExpSub(ExpSoma(ValorInteiro(4), ValorInteiro(12)), ValorInteiro(3))
        programa = Programa(expressao)
        if programa.checaTipo():
            programa.executar()
        else:
            print("Expressao mal tipada.")


if __name__ == "__main__":
    Exemplo1.main()
